### PR TITLE
feat(2692): create custom action for image scanning

### PR DESCRIPTION
**Pull request by golebu2020**

### CODE REVIEW

File: .github/actions/scan-docker-image/action.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    description: Path to Dockerfile
+    required: true
+  context:
+    description: Dockerfile context
+    required: true
+
+runs:
+  using: composite
+  steps:
+  - name: Build Docker Second Image
```
Comment (junminahn) [inline .github/actions/scan-docker-image/action.yml:20]: Why do we need to build the second image?

File: .github/actions/scan-docker-image/action.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    uses: docker/build-push-action@v4
+    with:
+      context: context
```
Comment (junminahn) [inline .github/actions/scan-docker-image/action.yml:23]: is this a typo?

File: .github/workflows/deploy-test.yml [hunk 3]
```diff
@@ -176,3 +176,40 @@ jobs:
           {
             "text": ":warning: Deployment to Test failed! [Investigate here](https://github.com/bcgov/platform-services-registry/)"
           }
+
+  scan-images-for-vulnerabilities:
+    needs: [deploy]
```
Comment (junminahn) [inline .github/workflows/deploy-test.yml:181]: we can run this job in parallel with the deployment job

File: .github/workflows/deploy-dev.yml [hunk 4]
```diff
@@ -218,3 +218,52 @@ jobs:
           {
             "text": ":warning: Deployment to Prod failed! [Investigate here](https://github.com/bcgov/platform-services-registry/)"
           }
+
+  scan-images-for-vulnerabilities:
+    needs: [deploy]
```
Comment (junminahn) [inline .github/workflows/deploy-dev.yml:223]: we can run this job in parallel with the deployment job

File: .github/workflows/deploy-dev.yml [hunk 5]
```diff
@@ -218,3 +218,52 @@ jobs:
...
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
```
Comment (junminahn) [inline .github/workflows/deploy-dev.yml:227]: are we writing packages?

File: .github/workflows/deploy-test.yml [hunk 6]
```diff
@@ -176,3 +176,40 @@ jobs:
...
+      security-events: write
+
+    steps:
+    - name: Log in to ghcr
+      run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+    - name: App
+      uses: ./.github/actions/scan-docker-image
+      with:
+        image: ${{ env.GITHUB_REGISTRY }}/bcgov/pltsvc:${{ github.sha }}
```
Comment (junminahn) [inline .github/workflows/deploy-test.yml:196]: `sha`?

File: .github/actions/scan-docker-image/action.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+runs:
+  using: composite
+  steps:
+  - name: Run Trivy vulnerability scanner
+    uses: aquasecurity/trivy-action@0.28.0
+    with:
+      image-ref: ${{ inputs.image }}
+      format: sarif
+      output: trivy-results-${{ inputs.image }}.sarif
```
Comment (junminahn) [inline .github/actions/scan-docker-image/action.yml:19]: `image` is a URL contains `/`, not sure that creating sub directory is ideal for this case; try without the prefix as each job runs one by one?

File: .github/actions/scan-docker-image/action.yml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+name: Trivy Scan Action
+description: Scan Docker images using Trivy and upload results.
+inputs:
+  image:
+    description: A docker image
+    required: true
+  category:
```
Comment (junminahn) [inline .github/actions/scan-docker-image/action.yml:7]: let's remove `category` and use `image` for `sarif` category.

Comment (golebu2020) [reply .github/actions/scan-docker-image/action.yml:7]: Okay

**Review by junminahn — APPROVED** [review]

**Pull request merged**